### PR TITLE
fix(GrapesJsBuilder): prevent CKEditor modal from closing when dragging text selection outside the dialog

### PR DESCRIPTION
**Pull request by edouard-mangel**
## Summary

Hi team 👋 — this PR fixes a UX issue in the CKEditor modal (used when editing text blocks in the GrapesJS email builder) that we expect would generate support requests from users if left unaddressed.

### The bug

When a user:
1. Opens the CKEditor modal to edit a text block
2. Selects text by clicking and dragging
3. Accidentally drags the mouse **outside** the dialog before releasing

…the modal **closes immediately and all unsaved content is lost**.

This is a particularly painful experience because text selection naturally involves imprecise mouse movements — especially on smaller screens or when selecting long passages. Losing a carefully written email body with a single accidental mouse release is the kind of thing that causes real frustration and can erode trust in the editor.

### Root cause

When the user releases the mouse outside the dialog, the browser fires a `click` event on the GrapesJS backdrop element (`.gjs-mdl-container`). GrapesJS cannot tell the difference between:
- a **genuine backdrop click** (intentional: close the modal), and
- a **drag that ended outside** (unintentional: user was just selecting text)

Both produce an identical `click` event on the backdrop.

### Fix

A small utility — `preventModalCloseOnDrag` — tracks where `mousedown` originated. If a `click` reaches the backdrop but the preceding `mousedown` was **inside** the dialog, the click is silently absorbed and the modal stays open. A genuine backdrop click (where `mousedown` also originated on the backdrop) still closes the modal as expected, so no existing behavior is changed.

### Changes

| File | What changed |
|------|-------------|
| `plugins/grapesjs.ckeditor.js` | Calls `preventModalCloseOnDrag` on modal open; cleans up on close |
| `plugins/preventModalCloseOnDrag.js` | New extracted utility (makes the logic independently testable) |
| `plugins/__tests__/preventModalCloseOnDrag.test.js` | 8 Jest tests covering the fix and edge cases |
| `package.json` / `.babelrc` | Added Jest + jsdom test infrastructure (first JS tests in this bundle) |

## Test plan

- [ ] Open the GrapesJS email builder and double-click a text block to open the CKEditor modal
- [ ] Type some content, then drag to select text — intentionally drag the mouse outside the dialog before releasing
- [ ] Verify the modal **stays open** and content is preserved
- [ ] Click directly on the dark backdrop (outside the dialog) — verify the modal **does close** as expected
- [ ] Run `npm test` inside `plugins/GrapesJsBuilderBundle` — all 8 tests should pass

---

> **Note:** The `package-lock.json` update reflects the addition of Jest and jsdom as dev dependencies. These are test-only and do not affect the production bundle.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/GrapesJsBuilderBundle/package.json [hunk 1]
```diff
@@ -62,12 +63,20 @@
...
+    "testEnvironment": "jsdom",
+    "transform": {
+      "^.+\\.js$": "babel-jest"
+    }
+  },
   "devDependencies": {
     "@babel/cli": "^7.16.8",
     "@babel/core": "^7.16.7",
     "@babel/plugin-proposal-class-properties": "^7.16.7",
+    "@babel/plugin-transform-modules-commonjs": "^7.28.6",
```
[the remark this answers is not in the document]
Comment (edouard-mangel) [reply plugins/GrapesJsBuilderBundle/package.json]: Done — aligned to `^7.16.7` to match the rest of the Babel toolchain. The version has been updated and the commit amended.

File: composer.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
     },
   "require": {
     "composer/installers": "^1.11",
-    "mautic/core-lib": "^5.0"
+    "mautic/core-lib": "^5.0",
+    "acquia/mc-cs-plugin-custom-objects": "dev-staging"
   },
```
[the remark this answers is not in the document]
Comment (edouard-mangel) [reply composer.json:26]: Agreed, this was an unrelated commit that had leaked in from our fork's `5.2` branch. The branch has been rebased directly onto `upstream/mautic:5.2`, removing the Composer dependency entirely from this PR. The composer.json and composer.lock changes will be handled in a dedicated PR.

File: composer.json [hunk 3]
```diff
@@ -106,6 +107,10 @@
     }
   },
   "repositories": [
+    {
+      "type": "vcs",
+      "url": "https://github.com/WebAnyOne/mc-cs-plugin-custom-objects"
+    },
```
[the remark this answers is not in the document]
Comment (edouard-mangel) [reply composer.json]: Removed — the branch has been rebased onto `upstream/mautic:5.2`, so neither this VCS repository entry nor the Custom Objects dependency appear in the diff anymore.

File: composer.json [hunk 4]
```diff
@@ -182,6 +187,9 @@
       "app": [
         "type:mautic-core"
       ],
+      "plugins/CustomObjectsBundle": [
+        "acquia/mc-cs-plugin-custom-objects"
+      ],
```
[the remark this answers is not in the document]
Comment (edouard-mangel) [reply composer.json]: Removed as well — same rebase. The installer-path mapping, the VCS repository, and the Composer require for `acquia/mc-cs-plugin-custom-objects` are all gone from this PR. They will be properly introduced in a separate, focused PR.

**Comment by escopecz** [comment]
Hi, Mautic 5.2 is closed for bug fixes. Please choose the branch 7.0 for bug fixes or 7.x for feature requests.
